nemo/src/io/formats/sparql.rs

```diff
@@ -114,52 +99,53 @@ impl FormatBuilder for SparqlBuilder {
         _tag: Self::Tag,
         parameters: &Parameters<SparqlBuilder>,
         _direction: Direction,
-    ) -> Result<(Self, Option<ResourceSpec>), ValidationErrorKind> {
+    ) -> Result<Self, ValidationErrorKind> {
         // Copied from DsvBuilder
         let value_formats = parameters
             .get_optional(SparqlParameter::Format)
-            .map(|value| DsvValueFormats::try_from(value).unwrap());
+            .map(|value| DsvValueFormats::try_from(value).expect("value formats have already been validated"));
 
         let limit = parameters
             .get_optional(SparqlParameter::Limit)
             .map(|value| value.to_u64_unchecked());
 
-        let ignore_headers = parameters
-            .get_optional(SparqlParameter::IgnoreHeaders)
-            .as_ref()
-            .map(AnyDataValue::to_boolean_unchecked)
-            .unwrap_or(false);
-
-        // Sparql-specific fields
+        // SPARQL-specific fields
         let endpoint = Iri::parse_unchecked(
             parameters
-                .get_required(SparqlParameter::Endpoint.into())
+                .get_required(SparqlParameter::Endpoint)
                 .to_iri_unchecked(),
         );
 
         let query = parameters
-            .get_required(SparqlParameter::Query.into())
+            .get_required(SparqlParameter::Query)
             .to_plain_string_unchecked();
 
-        let parsed_query = Query::parse(query.as_str(), None).unwrap();
+        let query = Query::parse(query.as_str(), None).expect("query has already been validated");
 
         // Create and return a new ResourceSpec to replace the existing ResourceSpec
-        let resource = ResourceSpec::from_endpoint(endpoint, query);
-        Ok((
-            Self {
-                delimiter: b'\t',
-                value_formats,
-                limit,
-                ignore_headers,
-                parsed_query,
-            },
-            Some(resource),
-        ))
+        Ok(Self {
+            value_formats,
+            limit,
+            endpoint,
+            query,
+        })
+    }
+
+    // TODO: do we need to parse parameters here again? that would be annoying
```

Comment:
We definitely shouldn't need to. All that should be needed here is to replace the IRI with one that also contains the query.